app/src/main/java/com/allan/boardbuddies/activities/CanvasActivity.java

```diff
@@ -61,67 +60,49 @@ public void onGlobalLayout() {
                 }
             });
         }
+
+        filePath = (File)getIntent().getExtras().get("FILEPATH");
+        if (getIntent().getExtras().get("FILENAME") != null){
+            localPosition = getIntent().getExtras().getInt("POSITION");
+            localFilename = getIntent().getExtras().getString("FILENAME");
+            localBoard = new Gson().fromJson(Utilities.getFileAsString(new File(filePath, localFilename)), Board.class);
+            localTitle = localBoard.getTitle();
+            canvasTitle.setText(localTitle);
+            Board tempBoard = new Gson().fromJson(Utilities.getFileAsString(new File(filePath, localFilename)), Board.class);
```

Comment:
doesn't this give the same thing as localBoard on line 68?

Reply:
no i need a deep copy to work with and this was the easiest way i could think of lol

Reply:
would recommend just implementing .copy() on Board class